src/main/kotlin/io/snyk/plugin/ui/toolwindow/SuggestionDescriptionPanel.kt

```diff
@@ -399,12 +402,40 @@
~                    addActionListener { e ->
~                        val dialog = ReportFalsePositiveDialog(
~                            project,
                             titlePanel(insets = JBUI.insetsBottom(10), indent = 0)
                         )
                         if (dialog.showAndGet()) {
-                            this.isEnabled = false
-                            this.text = FALSE_POSITIVE_REPORTED_TEXT
-                            // todo: disable re-report per session/project/application/token/org ?
+                            val suggestionRange = suggestion.ranges?.getOrNull(suggestionIndex) //fixme
+                            val snykApiService = service<SnykApiService>() //fixme
+                            val payload = FalsePositivePayload(
+                                topic = "False Positive",
+                                message = suggestion.message,
+                                context = FalsePositiveContext(
+                                    issueId = suggestion.id,
+                                    userPublicId = snykApiService.userId ?: "",
+                                    startLine = suggestionRange?.startRow ?: 1,
+                                    endLine = suggestionRange?.endRow ?: 1,
+                                    primaryFilePath = snykCodeFile.virtualFile.path,
+                                    vulnName = suggestion.rule,
+                                    fileContents = "dialog.result" //fixme
+                                )
+                            )
+                            if (snykApiService.reportFalsePositive(payload)) {
+                                this.isEnabled = false
```

Comment:
Would now be the time to update it?

Reply:
Will discuss with Andy. Imho the best way to inform user about succeed post - will be disable button with appropriate text. At least that how Ide do it (in Exception reporter for example)